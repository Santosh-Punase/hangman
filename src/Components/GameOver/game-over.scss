@import 'src/assets/scss/variables';

.hangman__game-over {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 100vh;
  background-color: $blue-dark;
  opacity: 0.8;
  z-index: 100;
  text-transform: uppercase;
  user-select: none;

  &__text {
    color: $white;
    font-size: 3rem;
    opacity: 1;
    margin-bottom: 1rem;
  }

  &__button {
    opacity: 1;
    font-size: 1rem;
    color: $orange;
    z-index: 150;
    padding: 0.5rem 1.5rem;
    border-radius: 1rem;
    border: 2px dotted $orange;
    cursor: pointer;
  }
}