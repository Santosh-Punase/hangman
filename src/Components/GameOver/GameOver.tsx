import React from 'react';

import './game-over.scss';

interface Props {
  isGameOver: boolean;
  hasPlayerWon: boolean;
  onNewWordClick: () => void;
}

const GameOver: React.FC<Props> = ({ onNewWordClick, isGameOver, hasPlayerWon }) => {

  if(isGameOver) {
    return (
      <div className="hangman__game-over">
        <div className="hangman__game-over__text">Game Over</div>
        <div className="hangman__game-over__button" onClick={onNewWordClick}>New word</div>
      </div>
    );
  }

  if(hasPlayerWon) {
    return (
      <div className="hangman__game-over">
        <div className="hangman__game-over__text">You won!!!</div>
        <div className="hangman__game-over__button" onClick={onNewWordClick}>Start over</div>
      </div>
    );
  }

  return null;
}

export { GameOver };
