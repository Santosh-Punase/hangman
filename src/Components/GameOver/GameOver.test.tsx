import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { GameOver } from './GameOver';

afterEach(cleanup);

const props = {
  onNewWordClick: jest.fn(),
  isGameOver: false,
  hasPlayerWon: false
};

describe('Game Over screen', () => {
  test('should Not render initially', () => {
    const { asFragment } = render(<GameOver { ...props } />);
    
    expect(asFragment()).toMatchInlineSnapshot(`<DocumentFragment />`);
  });

  test('should render gameover', () => {
    const { asFragment } = render(<GameOver { ...props } isGameOver />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render player Won', () => {
    const { asFragment } = render(<GameOver { ...props } hasPlayerWon />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  test('should start new game on "new word" click', () => {
    const { getByText } = render(<GameOver { ...props } isGameOver />);
    
    fireEvent.click(getByText(/New word/i));

    expect(props.onNewWordClick).toHaveBeenCalledTimes(1);
  });
});
