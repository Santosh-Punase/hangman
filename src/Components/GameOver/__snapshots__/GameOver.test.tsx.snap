// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Over screen should render gameover 1`] = `
<DocumentFragment>
  <div
    class="hangman__game-over"
  >
    <div
      class="hangman__game-over__text"
    >
      Game Over
    </div>
    <div
      class="hangman__game-over__button"
    >
      New word
    </div>
  </div>
</DocumentFragment>
`;

exports[`Game Over screen should render player Won 1`] = `
<DocumentFragment>
  <div
    class="hangman__game-over"
  >
    <div
      class="hangman__game-over__text"
    >
      You won!!!
    </div>
    <div
      class="hangman__game-over__button"
    >
      Start over
    </div>
  </div>
</DocumentFragment>
`;
